begin v3 era contributing documentation (#1177)
exercism/rust: docs/CONTRIBUTING.md

# Contributing to the Rust Exercism Track

Thank you so much for contributing! :tada:
We welcome pull requests of all kinds.
As one of the many tracks in Exercism, contributions here should observe Exercism standards like the [Code of Conduct](https://exercism.io/code-of-conduct).
This document articulates maintainers expectations of contributors.
It describes what is specific about working on the Rust track and how you can help.

## Ways to Contribute

- improving existing exercises
- creating new exercises
- internal tooling

## Merging Philosophy

A [pull request](https://docs.github.com/en/github/getting-started-with-github/github-glossary#pull-request) should address one logical change.
This could be small or big.

> For example, [#1174](https://github.com/exercism/rust/pull/1174) fixed a single typo in a single file after minutes of collaboration.

Reviewer: ```suggestion
For example, [#1175](https://github.com/exercism/rust/pull/1175) fixed a single typo in a single file after minutes of collaboration.
```

Fairly sure that was just an off-by-one error.

Author: Perfect, thank you!

# Contributing to the Rust Exercism Track

This document is a work-in-progress.

> Current maintainers are primarily preparing for the v3 launch of the Rust track.

Reviewer: This line will get stale fast. One way we might future-proof it is by phrasing the paragraph like:

This track is a work in progress. Please take all the value you can from it, but if you notice any way to improve it, we are eager for pull requests. Fixing a typo in documentation is every bit as valid a contribution as a massive addition of code.

> A work of art is never finished, merely abandoned.
>
> -- Paul Valéry

Author: I love the Paul Valéry quote. Frames our work quite wonderfully.

Nonetheless, feel free to peruse what we have written thus far!

*Contributions welcome :)*

> Thank you so much for contributing! :tada:

Reviewer: Though this line was inherited from previous documentation, I feel that it fits poorly here. Probably the best options might be to move it back to the first non-header line of the document, or remove it.

Author: Great point. I feel removing is best!

As one of the many tracks in Exercism, contributions here should observe Exercism standards like the [Code of Conduct](https://exercism.io/code-of-conduct).
This document introduces the ways you can help and what maintainers expect of contributors.

## Ways to Contribute

> As with many Open Source Projects, work abounds.

Reviewer: ```suggestion
As with many Open Source projects, work abounds.
```

I don't think that "Projects" deserves capitalization here; it's not really part of the phrase.

> Here are a few categories of what contributions:

Reviewer: ```suggestion
Here are a few categories of welcome contribution:
```

Clarify grammar.

Also might be worth adding:

- fixing typos, misspellings, and grammatical errors
- updating documentation
